webapp/src/components/Analysis/UtterancesTable.tsx

```diff
@@ -390,6 +414,89 @@ const UtterancesTable: React.FC<Props> = ({
   ];
 
   const searchString = constructSearchString(pipeline);
+
+  const removeSpecialCharacters = (text: string) =>
+    text.replace(/[^a-zA-Z0-9_]/g, "");
```

Comment:
Can you explain why we have to remove special characters? How about if the persistent ids do contain special characters? I think it will fail.
For your information, this regular expression seems to be equivalent to this shortcut:
```suggestion
    text.replace(/\W/g, "");
```